### ISSUE TITLE
Unable to deregister a service when using ACLs

### ISSUE THREAD

**Issue by sergiords** [spring-cloud/spring-cloud-consul#326]
When using a Consul configured with ACLs service gets properly registrated:
```sh
18:29:19.062 DEBUG org.apache.http.headers http-outgoing-2 >> PUT /v1/agent/service/register?token=xxxx-xxxx-xxxx-xxxxx HTTP/1.1
18:29:19.062 DEBUG org.apache.http.headers http-outgoing-2 >> Content-Length: 162
18:29:19.062 DEBUG org.apache.http.headers http-outgoing-2 >> Content-Type: text/plain; charset=UTF-8
18:29:19.062 DEBUG org.apache.http.headers http-outgoing-2 >> Host: localhost:8500
18:29:19.063 DEBUG org.apache.http.headers http-outgoing-2 >> Connection: Keep-Alive
18:29:19.063 DEBUG org.apache.http.headers http-outgoing-2 >> User-Agent: Apache-HttpClient/4.5.3 (Java/1.8.0_121)
18:29:19.063 DEBUG org.apache.http.headers http-outgoing-2 >> Accept-Encoding: gzip,deflate
18:29:19.065 DEBUG org.apache.http.headers http-outgoing-2 << HTTP/1.1 200 OK
```

However when application is shutdown service is not deregistrated due to lack of permission (token is not passed, neither in parameter, nor in header).

```sh
18:45:02.442 DEBUG org.apache.http.headers http-outgoing-2 >> GET /v1/agent/service/deregister/my-service HTTP/1.1
18:45:02.442 DEBUG org.apache.http.headers http-outgoing-2 >> Host: localhost:8500
18:45:02.442 DEBUG org.apache.http.headers http-outgoing-2 >> Connection: Keep-Alive
18:45:02.442 DEBUG org.apache.http.headers http-outgoing-2 >> User-Agent: Apache-HttpClient/4.5.3 (Java/1.8.0_121)
18:45:02.442 DEBUG org.apache.http.headers http-outgoing-2 >> Accept-Encoding: gzip,deflate
18:45:02.442 DEBUG org.apache.http.headers http-outgoing-2 << HTTP/1.1 403 Forbidden
```

Plus, client uses a GET method instead of a PUT method as documented in [Consul API](https://www.consul.io/api/agent/service.html#deregister-service)

**Comment by spencergibb** [comment]
@sergiords the client is a third part lib https://github.com/Ecwid/consul-api. Looks like changes will need to be made there.

**Comment by sergiords** [comment]
@spencergibb yes and there is an [open issue with a PR](https://github.com/Ecwid/consul-api/issues/111) for that.
I think there are still a few changes to be made in ConsulLifecycle.java when client is updated, just to pass the token from properties.

**Comment by spencergibb** [comment]
Yes, we'll keep this open until the client is updated.

**Comment by fedyafed** [comment]
Hi!
@sergiords, @spencergibb [The client issue with PR is closed](https://github.com/Ecwid/consul-api/issues/111) in 1.2.4.

**Comment by spencergibb** [comment]
Closed via #401
